lib/active_record/connection_adapters/firebird/database_statements.rb

```diff
@@ -50,6 +50,8 @@ def exec_query(sql, name = 'SQL', binds = [], prepare: false)
           else
             result
           end
+        rescue RangeError
+          ActiveRecord::Result.new([], [])
```

Comment:
Although I don't have access to this repo I would like to suggest a small improvement:

Is it possible to write a test so that we don't get a regression?

Reply:
Yes it is possible .. thanks for the suggestion